plugins/colorize/colorize.plugin.zsh

```diff
@@ -26,3 +30,25 @@ colorize_via_pygmentize() {
         fi
     done
 }
+
+colorize_via_pygmentize_less() (   
+    # this function is a subshell so tmp_files can be shared to cleanup function
+    declare -a tmp_files 
+    
+    cleanup () {
+        [[ "${#tmp_files[@]}" != "0" ]] && rm -f "${tmp_files[@]}"
+        if [ -n "$1" ]; then kill -$1 $$; fi #signal propagation
+    }
+    trap 'cleanup' EXIT
+    trap 'cleanup HUP' HUP
+    trap 'cleanup TERM' TERM
+    trap 'cleanup INT' INT
+    
+    while (( $# != 0 )); do     #TODO: filter out less opts
+        tmp_file="$(mktemp --tmpdir tmp.colorize.XXXX.$( sed -E 's/[^a-Z0-9]/./g' <<< "$1"))"
```

Comment:
My `sed` complains about an invalid range end, it should be `a-zA-Z`, not `a-Z`. Why are you doing this sed operation though?

Reply:
Ok, I made it more portable, like this:
`    sed -E 's/[^a-zA-Z0-9-]/./g'`
2 reasons: I made that sed to prevent problems with files that may include unwanted characters like 
`    mark's c**l file~`
to expand, or break quoting, so it replaces any non alphanumerical with a dot. (Do you think this is unnecesarry?)
Second reason, it also replaces directory separators /, so if I do something like
`cless $(find . -name pom.xml) # this work's with my shell opts`
I will be _cless_ing many files with the same name but I will be able to see the path of the file I'm reading.
For comparison, the less status at the bottom would be
`./main/dev/target/pom.xml (file 1 of 8) lines 1-36/151 29%`
and the cless status
`/tmp/tmp.colorize.BdN9...main.dev.target.pom.xml (file 1 of 8) lines 1-36/151 27%`
If I didn't do the sed and I wanted to be able to differentiate them, I would have to do `mkdir -p` on /tmp, which I think would be making too much dirt...

Reply:
I changed it so that it only replaces slashes in the filename, the rest of characters shouldn't be an issue since the code is properly quoted.